orchestrator/careplanservice/handle_get.go

```diff
@@ -0,0 +1,191 @@
+package careplanservice
+
+import (
+	"context"
+	"encoding/json"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/auth"
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/rs/zerolog/log"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"net/http"
+	"strings"
+)
+
+func (s *Service) handleGet(httpResponse http.ResponseWriter, request *http.Request) error {
+	queryParams := request.URL.Query()
+	urlParts := strings.Split(strings.TrimPrefix(request.URL.Path, basePath+"/"), "/")
+	switch urlParts[0] {
+	case "CarePlan":
+		// fetch CarePlan + CareTeam, validate requester is participant of CareTeam
+		// if there are query params present, we need to handle the request a bit differently
+		if len(queryParams) != 0 {
+			if queryParams.Get("_id") == "" {
```

Comment:
why would the caller be using search if they have the ID of the CarePlan they want to request? SInce then they could just be getting the CarePlan with this ID? (`CarePlan/(id)`), then they won't get a bundle in return (sounds easier)

Reply:
This actually came up due to a search we are doing in CPC:
```
carePlanId := strings.TrimPrefix(strings.TrimPrefix(u.Path, "/cps/CarePlan/"), s.carePlanServiceURL.String())
	err = carePlanServiceClient.Read("CarePlan", &bundle, fhirclient.QueryParam("_id", carePlanId), fhirclient.QueryParam("_include", "CarePlan:care-team"))
	if err != nil {
		return err
	}
```

The CareTeam is required to check if the participant has a valid period

Reply:
I think I can modify this so we don't have to worry about _id being passed in, will test my idea out